### PR TITLE
Canvas: Improve resource picker initialization

### LINKED ISSUE #57311: Canvas: Improve resource picker UX
We currently default to setting initial screen of resource picker to folder view.

It may make more sense to make default be the url picker instead given the limited set of background images available in folder view (if available at all, see [here](https://github.com/grafana/grafana/discussions/56835#discussioncomment-3917279)).

### PR DESCRIPTION
**Pull request by nmarrs**
**What this PR does / why we need it**:
Improve resource picker initialization based on context of current value. If value is a url, initialize in the URL tab else initialize on the folder tab.

https://user-images.githubusercontent.com/22381771/197621208-833b5752-4153-4fb8-af52-209355afa2d7.mov

**Which issue(s) this PR fixes**:

Fixes #57311

### CODE REVIEW

File: public/app/plugins/panel/canvas/editor/options.ts [hunk 1]
```diff
@@ -39,6 +40,7 @@ export const optionBuilder: OptionSuppliers = {
         editor: ResourceDimensionEditor,
         settings: {
           resourceType: 'image',
+          defaultTab: PickerTabType.URL,
```
Comment (ryantxu) [inline public/app/plugins/panel/canvas/editor/options.ts:43]: 🤔 I don't think we really want URL to be the default, just that if an URL is specified it is shown as the option

File: public/app/features/dimensions/editors/ResourceDimensionEditor.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ const dummyFieldSettings: StandardEditorsRegistryItem<string, FieldNamePickerCon
 } as any;
 
 export const ResourceDimensionEditor: FC<
```
Comment (ryantxu) [inline public/app/features/dimensions/editors/ResourceDimensionEditor.tsx:28]:
```suggestion
export function ResourceDimensionEditor(props: StandardEditorProps<ResourceDimensionConfig, ResourceDimensionOptions, unknown>) {
```

lets avoid FC when we find them...

File: public/app/plugins/panel/canvas/editor/options.ts (hunk 1 above)
Comment (nmarrs) [reply public/app/plugins/panel/canvas/editor/options.ts:43]: In this case the URL is the default option only for the root background image for canvas. If we set logic to show URL tab based on if a url exists that messes up the resource picker in the context of picking icons as we set a url by default for any selected icon
Screen Shot 2022-10-20 at 10 25 46 AM

Comment (ryantxu) [reply public/app/plugins/panel/canvas/editor/options.ts:43]: but the URL we set is one that is known to us right?  it is relative to our root and actually points to a folder we know about.

the URL will start with `http://...` i think

File: public/app/features/dimensions/editors/ResourcePickerPopover.tsx [hunk 3]
```diff
@@ -37,8 +37,9 @@ export const ResourcePickerPopover = (props: Props) => {
   const { dialogProps } = useDialog({}, ref);
   const { overlayProps } = useOverlay({ onClose, isDismissable: true, isOpen: true }, ref);
 
+  const isURL = value && value.includes('://');
```
Comment (ryantxu) [inline public/app/features/dimensions/editors/ResourcePickerPopover.tsx:40]: :)

**Review by ryantxu — APPROVED** [review]

**Pull request merged**